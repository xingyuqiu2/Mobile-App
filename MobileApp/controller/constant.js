import TOKEN from './token';

let username = 'xingyuq2';

export const getName = () => username;

export const setName = (name) => {
  username = name;
};

export const BASE_URL = 'https://api.github.com/graphql';

export const HEADERS = {
  'Content-Type': 'application/json',
  Authorization: `bearer ${TOKEN}`,
};

// graphQL query to get information in profile screen
export const queryPr = (userName) => ({
  query: `query { 
            user(login: "${userName}") {
              avatarUrl
              name
              login
              bio
              websiteUrl
              email
              repositories {
                totalCount
              }
              following {
                totalCount
              }
              followers {
                totalCount
              }
              createdAt
            }
          }`,
});

// graphQL query to get information in repositories screen
export const queryRe = (userName) => ({
  query: `query { 
      user(login:"${userName}") {
        repositories {
          totalCount
        }
        repositories {
          nodes{
            name
            owner {
              login
            }
            description
          }
        }
      }
    }`,
});

// graphQL query to get information in followers and following screen
export const getQueryFollow = (userName, type) => ({
  query: `query { 
    user(login:"${userName}") {
      ${type} {
        totalCount
      }
      ${type} {
        nodes{
          avatarUrl
          name
          login
        }
      }
    }
  }`,
});
