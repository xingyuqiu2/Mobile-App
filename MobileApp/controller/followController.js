/* eslint-disable no-undef */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/react-in-jsx-scope */
import React, { useEffect, useState } from 'react';
import {
  StyleSheet, Text, SafeAreaView, FlatList, Image, TouchableOpacity,
} from 'react-native';
import 'node-fetch';
import Follow from '../model/follow';
import {
  setName, BASE_URL, HEADERS, getQueryFollow, getName,
} from './constant';

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    backgroundColor: '#d6dce7',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: '#3d322d',
    fontSize: 18,
    marginHorizontal: 15,
    marginVertical: 6,
  },
  touchableText: {
    color: '#2a683a',
    fontSize: 18,
    marginHorizontal: 15,
    marginVertical: 6,
  },
  itemWrapperStyle: {
    backgroundColor: '#f3e3c2',
    flexDirection: 'column',
    paddingHorizontal: 16,
    paddingVertical: 16,
    marginBottom: 5,
    marginTop: 5,
  },
  image: {
    width: 80,
    height: 80,
    marginTop: 5,
    marginBottom: 5,
    marginLeft: 5,
  },
});

/**
 * Fetch followers/following data using graphQL query.
 * @param {boolean} isMounted
 * @param {string} type followers/following
 * @returns response from fetch
 */
export async function fetchFollow(isMounted, type) {
  if (isMounted) {
    return fetch(BASE_URL, {
      method: 'POST',
      headers: HEADERS,
      body: JSON.stringify(getQueryFollow(getName(), type)),
    }).then((res) => res.json());
  }
  return null;
}

/**
 * Call fetchFollow and get data.
 * Then insert value from retreived data to content and return it.
 * @returns loaded followers/following content
 */
export default function GetFollow(navigation, type) {
  const [data, setData] = useState(null);

  useEffect(() => {
    let isMounted = true;
    async function getData() {
      const dataReceived = await fetchFollow(isMounted, type);
      setData(dataReceived);
    }
    getData();
    return () => { isMounted = false; };
  }, [type]);

  // handle invalid data
  if (!data) {
    return null;
  }
  // handle error
  if (!data.data.user) {
    return (
      <SafeAreaView style={styles.screen}>
        <Text>Not Found</Text>
      </SafeAreaView>
    );
  }

  // create follow object by passing part of a response
  const myFollow = new Follow(data.data.user[type].nodes);

  const listFollow = [];
  for (let i = 0; i < myFollow.avatarUrls.length; i += 1) {
    listFollow.push(
      {
        avatarUrl: myFollow.avatarUrls[i],
        name: myFollow.names[i],
        username: myFollow.usernames[i],
      },
    );
  }

  return (
    <SafeAreaView style={styles.screen}>
      <FlatList
        data={listFollow}
        keyExtractor={(item) => item.name}
        contentContainerStyle={{
          padding: 20,
        }}
        renderItem={({ item }) => (
          <SafeAreaView style={{
            backgroundColor: '#f3e3c2',
            flexDirection: 'row',
            paddingHorizontal: 16,
            paddingVertical: 16,
            marginBottom: 5,
            marginTop: 5,
          }}
          >
            <Image source={{ uri: item.avatarUrl }} style={styles.image} />
            <SafeAreaView style={styles.itemWrapperStyle}>
              <Text style={styles.text}>
                Name:
                {item.name}
              </Text>
              <TouchableOpacity onPress={() => {
                setName(item.username);
                navigation.push('Profile');
              }}
              >
                <Text style={styles.touchableText}>
                  Username:
                  {item.username}
                </Text>
              </TouchableOpacity>
            </SafeAreaView>
          </SafeAreaView>
        )}
      />
    </SafeAreaView>
  );
}
