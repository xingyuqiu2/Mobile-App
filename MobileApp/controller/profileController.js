/* eslint-disable no-undef */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/react-in-jsx-scope */
import React, { useEffect, useState } from 'react';
import {
  StyleSheet, Text, SafeAreaView, ScrollView, Image, TouchableOpacity,
} from 'react-native';
import PropTypes from 'prop-types';
import 'node-fetch';
import Profile from '../model/profile';
import {
  getName, setName, BASE_URL, HEADERS, queryPr,
} from './constant';

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    backgroundColor: '#d6dce7',
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    width: 150,
    height: 150,
    marginTop: 20,
    marginBottom: 10,
  },
  text: {
    color: '#3d322d',
    fontSize: 18,
    marginHorizontal: 15,
    marginVertical: 6,
  },
  touchableText: {
    color: '#3829d7',
    fontSize: 18,
    marginHorizontal: 15,
    marginVertical: 6,
  },
});

/**
 * Fetch profile data using graphQL query.
 * @param {boolean} isMounted
 * @returns response from fetch
 */
export async function fetchProfile(isMounted) {
  if (isMounted) {
    return fetch(BASE_URL, {
      method: 'POST',
      headers: HEADERS,
      body: JSON.stringify(queryPr(getName())),
    }).then((res) => res.json());
  }
  return null;
}

/**
 * Call fetchProfile and get data.
 * Then insert value from retreived data to content and return it.
 * @returns loaded profile content
 */
export default function GetProfile(navigation) {
  const [data, setData] = useState(null);

  useEffect(() => {
    let isMounted = true;
    async function getData() {
      const dataReceived = await fetchProfile(isMounted);
      setData(dataReceived);
    }
    getData();
    return () => { isMounted = false; };
  }, []);

  // handle invalid data
  if (!data) {
    return null;
  }
  // handle error
  if (!data.data.user) {
    return (
      <SafeAreaView style={styles.screen}>
        <Text>Not Found</Text>
      </SafeAreaView>
    );
  }

  // create profile object by passing part of a response
  const myProfile = new Profile(data.data.user);

  return (
    <SafeAreaView style={styles.screen}>
      <ScrollView contentContainerStyle={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
        <Image source={{ uri: myProfile.avatarUrl }} style={styles.image} />

        <Text style={styles.text}>{myProfile.name}</Text>
        <Text style={styles.text}>
          Username:
          {' '}
          {myProfile.userName}
        </Text>
        <Text style={styles.text}>
          Bio:
          {' '}
          {myProfile.bio}
        </Text>
        <Text style={styles.text}>
          Website:
          {' '}
          {myProfile.websiteUrl}
        </Text>
        <Text style={styles.text}>
          Email:
          {' '}
          {myProfile.email}
        </Text>

        <TouchableOpacity onPress={() => {
          setName(myProfile.userName);
          navigation.push('Repositories');
        }}
        >
          <Text style={styles.touchableText}>
            {myProfile.reposCount}
            {' '}
            repos
          </Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => {
          setName(myProfile.userName);
          navigation.push('Followers');
        }}
        >
          <Text style={styles.touchableText}>
            Followers:
            {' '}
            {myProfile.followersCount}
          </Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => {
          setName(myProfile.userName);
          navigation.push('Following');
        }}
        >
          <Text style={styles.touchableText}>
            Following:
            {' '}
            {myProfile.followingCount}
          </Text>
        </TouchableOpacity>

        <Text style={styles.text}>
          Creation date:
          {' '}
          {myProfile.createdAt}
        </Text>
      </ScrollView>
    </SafeAreaView>
  );
}

GetProfile.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};
