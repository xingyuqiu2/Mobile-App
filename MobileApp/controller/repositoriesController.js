/* eslint-disable no-undef */
/* eslint-disable react/jsx-filename-extension */
import React, { useEffect, useState } from 'react';
import {
  StyleSheet, Text, SafeAreaView, FlatList, TouchableOpacity,
} from 'react-native';
import 'node-fetch';
import Repository from '../model/repository';
import {
  getName, setName, BASE_URL, HEADERS, queryRe,
} from './constant';

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    backgroundColor: '#d6dce7',
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    color: '#3d322d',
    fontSize: 18,
    marginHorizontal: 15,
    marginVertical: 6,
  },
  repoName: {
    color: '#8e1912',
    fontSize: 18,
    marginHorizontal: 15,
    marginVertical: 6,
  },
  touchableText: {
    color: '#711674',
    fontSize: 18,
    marginHorizontal: 15,
    marginVertical: 6,
  },
  itemWrapperStyle: {
    backgroundColor: '#f3e3c2',
    flexDirection: 'column',
    paddingHorizontal: 16,
    paddingVertical: 16,
    marginBottom: 5,
    marginTop: 5,
  },
});

/**
 * Fetch repositories data using graphQL query.
 * @param {boolean} isMounted
 * @returns response from fetch
 */
export async function fetchRepositories(isMounted) {
  if (isMounted) {
    return fetch(BASE_URL, {
      method: 'POST',
      headers: HEADERS,
      body: JSON.stringify(queryRe(getName())),
    }).then((res) => res.json());
  }
  return null;
}

/**
 * Call fetchRepositories and get data.
 * Then insert value from retreived data to content and return it.
 * @returns loaded repository content
 */
export default function GetRepositories(navigation) {
  const [data, setData] = useState(null);

  useEffect(() => {
    let isMounted = true;
    async function getData() {
      const dataReceived = await fetchRepositories(isMounted);
      setData(dataReceived);
    }
    getData();
    return () => { isMounted = false; };
  }, []);

  // handle invalid data
  if (!data) {
    return null;
  }
  // handle error
  if (!data.data.user) {
    return (
      <SafeAreaView style={styles.screen}>
        <Text>Not Found</Text>
      </SafeAreaView>
    );
  }

  // create repository object by passing part of a response
  const myRepositories = new Repository(data.data.user.repositories.nodes);

  const listRepos = [];
  for (let i = 0; i < myRepositories.repoNames.length; i += 1) {
    listRepos.push(
      {
        name: myRepositories.repoNames[i],
        user: myRepositories.users[i],
        description: myRepositories.descriptions[i],
      },
    );
  }

  return (
    <SafeAreaView style={styles.screen}>
      <FlatList
        data={listRepos}
        keyExtractor={(item) => item.name}
        contentContainerStyle={{
          padding: 20,
        }}
        renderItem={({ item }) => (
          <SafeAreaView style={styles.itemWrapperStyle}>
            <Text style={styles.repoName}>
              Repo name:
              {item.name}
            </Text>
            <TouchableOpacity onPress={() => {
              setName(item.user);
              navigation.push('Profile');
            }}
            >
              <Text style={styles.touchableText}>
                Username:
                {item.user}
              </Text>
            </TouchableOpacity>
            <Text style={styles.text}>
              Description:
              {item.description}
            </Text>
          </SafeAreaView>
        )}
      />
    </SafeAreaView>
  );
}
