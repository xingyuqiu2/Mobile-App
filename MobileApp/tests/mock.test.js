/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/react-in-jsx-scope */
import React from 'react';
import { act } from 'react-dom/test-utils';
import { render, screen, waitForElementToBeRemoved } from '@testing-library/react';
import { fetchProfile } from '../controller/profileController';
import { fetchRepositories } from '../controller/repositoriesController';
import { fetchFollow } from '../controller/followController';
import {
  ProfileScreen, RepositoriesScreen, FollowersScreen, FollowingScreen,
} from '../view/app';

const dataPr = {
  data: {
    user: {
      avatarUrl: 'https://avatars.githubusercontent.com/u/42951409?v=4',
      name: 'Jerry',
      login: 'xingyuq2',
      bio: 'Senior student',
      websiteUrl: 'https://github.com/xingyuqiu2',
      email: 'xxx@gmail.com',
      repositories: {
        totalCount: 6,
      },
      following: {
        totalCount: 5,
      },
      followers: {
        totalCount: 4,
      },
      createdAt: '2018-09-04T01:11:54Z',
    },
  },
};

const dataRe = {
  data: {
    user: {
      repositories: {
        totalCount: 3,
        nodes: [
          {
            name: 'Lab1',
            owner: {
              login: 'xyz',
            },
            description: 'Lab 1 starter code.',
          },
          {
            name: 'Lab2',
            owner: {
              login: 'qwe',
            },
            description: 'Lab2 code',
          },
          {
            name: 'Lab3',
            owner: {
              login: 'haha',
            },
            description: 'Fall 2018 Lab3 starter code.',
          },
        ],
      },
    },
  },
};

const dataFollowing = {
  data: {
    user: {
      following: {
        totalCount: 3,
        nodes: [
          {
            avatarUrl: 'https://avatars.githubusercontent.com/u/12994887?u=6bfec84cb512892557cfed7fd7c52b0b0f41f95b&v=4',
            name: 'Tom',
            login: 'tom2',
          },
          {
            avatarUrl: 'https://avatars.githubusercontent.com/u/12994887?u=6bfec84cb512892557cfed7fd7c52b0b0f41f95b&v=4',
            name: 'Jerry',
            login: 'jerry2',
          },
          {
            avatarUrl: 'https://avatars.githubusercontent.com/u/12994887?u=6bfec84cb512892557cfed7fd7c52b0b0f41f95b&v=4',
            name: '稚晖',
            login: 'peng-zhihui',
          },
        ],
      },
    },
  },
};

const dataFollowers = {
  data: {
    user: {
      followers: {
        totalCount: 2,
        nodes: [
          {
            avatarUrl: 'https://avatars.githubusercontent.com/u/12994887?u=6bfec84cb512892557cfed7fd7c52b0b0f41f95b&v=4',
            name: 'Max',
            login: 'max2',
          },
          {
            avatarUrl: 'https://avatars.githubusercontent.com/u/12994887?u=6bfec84cb512892557cfed7fd7c52b0b0f41f95b&v=4',
            name: 'Min',
            login: 'min2',
          },
        ],
      },
    },
  },
};

const dataError = {
  data: {
    user: null,
  },
  errors: [
    {
      type: 'NOT_FOUND',
      path: [
        'user',
      ],
      locations: [
        {
          line: 7,
          column: 5,
        },
      ],
      message: "Could not resolve to a User with the login of 'cs125-student'.",
    },
  ],
};

beforeEach(() => {
  global.fetch.resetMocks();
});

test('returns result from fetchProfile if valid data', () => {
  global.fetch.mockResponseOnce(JSON.stringify(dataPr));
  const onResponse = jest.fn();
  const onError = jest.fn();

  return fetchProfile(true)
    .then(onResponse)
    .catch(onError)
    .finally(() => {
      expect(onResponse).toHaveBeenCalled();
      expect(onError).not.toHaveBeenCalled();

      expect(onResponse.mock.calls[0][0]).toEqual(dataPr);
    });
});

test('Matches loaded profile screen snapshot', async () => {
  await act(async () => {
    global.fetch.mockResponseOnce(JSON.stringify(dataPr));
    const { asFragment } = render(<ProfileScreen />);
    await waitForElementToBeRemoved(screen.getByText('x repos'));
    expect(asFragment()).toMatchSnapshot();
  });
});

test('Matches profile screen under error snapshot', async () => {
  await act(async () => {
    global.fetch.mockResponseOnce(JSON.stringify(dataError));
    const { asFragment } = render(<ProfileScreen />);
    await waitForElementToBeRemoved(screen.getByText('x repos'));
    expect(asFragment()).toMatchSnapshot();
  });
});

test('returns result from fetchRepositories if valid data', () => {
  global.fetch.mockResponseOnce(JSON.stringify(dataRe));
  const onResponse = jest.fn();
  const onError = jest.fn();

  return fetchRepositories(true)
    .then(onResponse)
    .catch(onError)
    .finally(() => {
      expect(onResponse).toHaveBeenCalled();
      expect(onError).not.toHaveBeenCalled();

      expect(onResponse.mock.calls[0][0]).toEqual(dataRe);
    });
});

test('Matches loaded repositories screen snapshot', async () => {
  await act(async () => {
    global.fetch.mockResponseOnce(JSON.stringify(dataRe));
    const { asFragment } = render(<RepositoriesScreen />);
    await waitForElementToBeRemoved(screen.getByText('Repo name:Loading'));
    expect(asFragment()).toMatchSnapshot();
  });
});

test('Matches repositories screen under error snapshot', async () => {
  await act(async () => {
    global.fetch.mockResponseOnce(JSON.stringify(dataError));
    const { asFragment } = render(<RepositoriesScreen />);
    await waitForElementToBeRemoved(screen.getByText('Repo name:Loading'));
    expect(asFragment()).toMatchSnapshot();
  });
});

test('returns result from fetchFollowing if valid data', () => {
  global.fetch.mockResponseOnce(JSON.stringify(dataFollowing));
  const onResponse = jest.fn();
  const onError = jest.fn();

  return fetchFollow(true, 'following')
    .then(onResponse)
    .catch(onError)
    .finally(() => {
      expect(onResponse).toHaveBeenCalled();
      expect(onError).not.toHaveBeenCalled();

      expect(onResponse.mock.calls[0][0]).toEqual(dataFollowing);
    });
});

test('Matches loaded following screen snapshot', async () => {
  await act(async () => {
    global.fetch.mockResponseOnce(JSON.stringify(dataFollowing));
    const { asFragment } = render(<FollowingScreen />);
    await waitForElementToBeRemoved(screen.getByText('Name:Loading'));
    expect(asFragment()).toMatchSnapshot();
  });
});

test('Matches following screen under error snapshot', async () => {
  await act(async () => {
    global.fetch.mockResponseOnce(JSON.stringify(dataError));
    const { asFragment } = render(<FollowingScreen />);
    await waitForElementToBeRemoved(screen.getByText('Name:Loading'));
    expect(asFragment()).toMatchSnapshot();
  });
});

test('returns result from fetchFollowers if valid data', () => {
  global.fetch.mockResponseOnce(JSON.stringify(dataFollowers));
  const onResponse = jest.fn();
  const onError = jest.fn();

  return fetchFollow(true, 'followers')
    .then(onResponse)
    .catch(onError)
    .finally(() => {
      expect(onResponse).toHaveBeenCalled();
      expect(onError).not.toHaveBeenCalled();

      expect(onResponse.mock.calls[0][0]).toEqual(dataFollowers);
    });
});

test('Matches loaded followers screen snapshot', async () => {
  await act(async () => {
    global.fetch.mockResponseOnce(JSON.stringify(dataFollowers));
    const { asFragment } = render(<FollowersScreen />);
    await waitForElementToBeRemoved(screen.getByText('Name:Loading'));
    expect(asFragment()).toMatchSnapshot();
  });
});

test('Matches followers screen under error snapshot', async () => {
  await act(async () => {
    global.fetch.mockResponseOnce(JSON.stringify(dataError));
    const { asFragment } = render(<FollowersScreen />);
    await waitForElementToBeRemoved(screen.getByText('Name:Loading'));
    expect(asFragment()).toMatchSnapshot();
  });
});
