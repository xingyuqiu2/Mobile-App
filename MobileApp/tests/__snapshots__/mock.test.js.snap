// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches followers screen under error snapshot 1`] = `
<DocumentFragment>
  <rctsafeareaview
    style="flex: 1; background-color: rgb(214, 220, 231); justify-content: center; align-items: center;"
  >
    <text>
      Not Found
    </text>
  </rctsafeareaview>
</DocumentFragment>
`;

exports[`Matches following screen under error snapshot 1`] = `
<DocumentFragment>
  <rctsafeareaview
    style="flex: 1; background-color: rgb(214, 220, 231); justify-content: center; align-items: center;"
  >
    <text>
      Not Found
    </text>
  </rctsafeareaview>
</DocumentFragment>
`;

exports[`Matches loaded followers screen snapshot 1`] = `
<DocumentFragment>
  <rctsafeareaview
    style="flex: 1; background-color: rgb(214, 220, 231); justify-content: center; align-items: center;"
  >
    <rctscrollview
      contentcontainerstyle="[object Object]"
      data="[object Object],[object Object]"
      initialnumtorender="10"
      maxtorenderperbatch="10"
      scrolleventthrottle="50"
      stickyheaderindices=""
      updatecellsbatchingperiod="50"
      viewabilityconfigcallbackpairs=""
      windowsize="21"
    >
      <view>
        <view>
          <rctsafeareaview
            style="background-color: rgb(243, 227, 194); flex-direction: row; margin-bottom: 5px; margin-top: 5px;"
          >
            <img
              source="[object Object]"
              style="width: 80px; height: 80px; margin-top: 5px; margin-bottom: 5px; margin-left: 5px;"
            />
            <rctsafeareaview
              style="background-color: rgb(243, 227, 194); flex-direction: column; margin-bottom: 5px; margin-top: 5px;"
            >
              <text
                style="color: rgb(61, 50, 45); font-size: 18px;"
              >
                Name:Max
              </text>
              <view
                focusable="true"
                style="opacity: 1;"
              >
                <text
                  style="color: rgb(42, 104, 58); font-size: 18px;"
                >
                  Username:max2
                </text>
              </view>
            </rctsafeareaview>
          </rctsafeareaview>
        </view>
        <view>
          <rctsafeareaview
            style="background-color: rgb(243, 227, 194); flex-direction: row; margin-bottom: 5px; margin-top: 5px;"
          >
            <img
              source="[object Object]"
              style="width: 80px; height: 80px; margin-top: 5px; margin-bottom: 5px; margin-left: 5px;"
            />
            <rctsafeareaview
              style="background-color: rgb(243, 227, 194); flex-direction: column; margin-bottom: 5px; margin-top: 5px;"
            >
              <text
                style="color: rgb(61, 50, 45); font-size: 18px;"
              >
                Name:Min
              </text>
              <view
                focusable="true"
                style="opacity: 1;"
              >
                <text
                  style="color: rgb(42, 104, 58); font-size: 18px;"
                >
                  Username:min2
                </text>
              </view>
            </rctsafeareaview>
          </rctsafeareaview>
        </view>
        <view
          style="height: 0px;"
        />
      </view>
    </rctscrollview>
  </rctsafeareaview>
</DocumentFragment>
`;

exports[`Matches loaded following screen snapshot 1`] = `
<DocumentFragment>
  <rctsafeareaview
    style="flex: 1; background-color: rgb(214, 220, 231); justify-content: center; align-items: center;"
  >
    <rctscrollview
      contentcontainerstyle="[object Object]"
      data="[object Object],[object Object],[object Object]"
      initialnumtorender="10"
      maxtorenderperbatch="10"
      scrolleventthrottle="50"
      stickyheaderindices=""
      updatecellsbatchingperiod="50"
      viewabilityconfigcallbackpairs=""
      windowsize="21"
    >
      <view>
        <view>
          <rctsafeareaview
            style="background-color: rgb(243, 227, 194); flex-direction: row; margin-bottom: 5px; margin-top: 5px;"
          >
            <img
              source="[object Object]"
              style="width: 80px; height: 80px; margin-top: 5px; margin-bottom: 5px; margin-left: 5px;"
            />
            <rctsafeareaview
              style="background-color: rgb(243, 227, 194); flex-direction: column; margin-bottom: 5px; margin-top: 5px;"
            >
              <text
                style="color: rgb(61, 50, 45); font-size: 18px;"
              >
                Name:Tom
              </text>
              <view
                focusable="true"
                style="opacity: 1;"
              >
                <text
                  style="color: rgb(42, 104, 58); font-size: 18px;"
                >
                  Username:tom2
                </text>
              </view>
            </rctsafeareaview>
          </rctsafeareaview>
        </view>
        <view>
          <rctsafeareaview
            style="background-color: rgb(243, 227, 194); flex-direction: row; margin-bottom: 5px; margin-top: 5px;"
          >
            <img
              source="[object Object]"
              style="width: 80px; height: 80px; margin-top: 5px; margin-bottom: 5px; margin-left: 5px;"
            />
            <rctsafeareaview
              style="background-color: rgb(243, 227, 194); flex-direction: column; margin-bottom: 5px; margin-top: 5px;"
            >
              <text
                style="color: rgb(61, 50, 45); font-size: 18px;"
              >
                Name:Jerry
              </text>
              <view
                focusable="true"
                style="opacity: 1;"
              >
                <text
                  style="color: rgb(42, 104, 58); font-size: 18px;"
                >
                  Username:jerry2
                </text>
              </view>
            </rctsafeareaview>
          </rctsafeareaview>
        </view>
        <view>
          <rctsafeareaview
            style="background-color: rgb(243, 227, 194); flex-direction: row; margin-bottom: 5px; margin-top: 5px;"
          >
            <img
              source="[object Object]"
              style="width: 80px; height: 80px; margin-top: 5px; margin-bottom: 5px; margin-left: 5px;"
            />
            <rctsafeareaview
              style="background-color: rgb(243, 227, 194); flex-direction: column; margin-bottom: 5px; margin-top: 5px;"
            >
              <text
                style="color: rgb(61, 50, 45); font-size: 18px;"
              >
                Name:稚晖
              </text>
              <view
                focusable="true"
                style="opacity: 1;"
              >
                <text
                  style="color: rgb(42, 104, 58); font-size: 18px;"
                >
                  Username:peng-zhihui
                </text>
              </view>
            </rctsafeareaview>
          </rctsafeareaview>
        </view>
        <view
          style="height: 0px;"
        />
      </view>
    </rctscrollview>
  </rctsafeareaview>
</DocumentFragment>
`;

exports[`Matches loaded profile screen snapshot 1`] = `
<DocumentFragment>
  <rctsafeareaview
    style="flex: 1; background-color: rgb(214, 220, 231); justify-content: center; align-items: center;"
  >
    <rctscrollview
      contentcontainerstyle="[object Object]"
    >
      <view>
        <img
          source="[object Object]"
          style="width: 150px; height: 150px; margin-top: 20px; margin-bottom: 10px;"
        />
        <text
          style="color: rgb(61, 50, 45); font-size: 18px;"
        >
          Jerry
        </text>
        <text
          style="color: rgb(61, 50, 45); font-size: 18px;"
        >
          Username: xingyuq2
        </text>
        <text
          style="color: rgb(61, 50, 45); font-size: 18px;"
        >
          Bio: Senior student
        </text>
        <text
          style="color: rgb(61, 50, 45); font-size: 18px;"
        >
          Website: https://github.com/xingyuqiu2
        </text>
        <text
          style="color: rgb(61, 50, 45); font-size: 18px;"
        >
          Email: xxx@gmail.com
        </text>
        <view
          focusable="true"
          style="opacity: 1;"
        >
          <text
            style="color: rgb(56, 41, 215); font-size: 18px;"
          >
            6 repos
          </text>
        </view>
        <view
          focusable="true"
          style="opacity: 1;"
        >
          <text
            style="color: rgb(56, 41, 215); font-size: 18px;"
          >
            Followers: 4
          </text>
        </view>
        <view
          focusable="true"
          style="opacity: 1;"
        >
          <text
            style="color: rgb(56, 41, 215); font-size: 18px;"
          >
            Following: 5
          </text>
        </view>
        <text
          style="color: rgb(61, 50, 45); font-size: 18px;"
        >
          Creation date: 2018-09-04T01:11:54Z
        </text>
      </view>
    </rctscrollview>
  </rctsafeareaview>
</DocumentFragment>
`;

exports[`Matches loaded repositories screen snapshot 1`] = `
<DocumentFragment>
  <rctsafeareaview
    style="flex: 1; background-color: rgb(214, 220, 231); justify-content: center; align-items: center;"
  >
    <rctscrollview
      contentcontainerstyle="[object Object]"
      data="[object Object],[object Object],[object Object]"
      initialnumtorender="10"
      maxtorenderperbatch="10"
      scrolleventthrottle="50"
      stickyheaderindices=""
      updatecellsbatchingperiod="50"
      viewabilityconfigcallbackpairs=""
      windowsize="21"
    >
      <view>
        <view>
          <rctsafeareaview
            style="background-color: rgb(243, 227, 194); flex-direction: column; margin-bottom: 5px; margin-top: 5px;"
          >
            <text
              style="color: rgb(142, 25, 18); font-size: 18px;"
            >
              Repo name:Lab1
            </text>
            <view
              focusable="true"
              style="opacity: 1;"
            >
              <text
                style="color: rgb(113, 22, 116); font-size: 18px;"
              >
                Username:xyz
              </text>
            </view>
            <text
              style="color: rgb(61, 50, 45); font-size: 18px;"
            >
              Description:Lab 1 starter code.
            </text>
          </rctsafeareaview>
        </view>
        <view>
          <rctsafeareaview
            style="background-color: rgb(243, 227, 194); flex-direction: column; margin-bottom: 5px; margin-top: 5px;"
          >
            <text
              style="color: rgb(142, 25, 18); font-size: 18px;"
            >
              Repo name:Lab2
            </text>
            <view
              focusable="true"
              style="opacity: 1;"
            >
              <text
                style="color: rgb(113, 22, 116); font-size: 18px;"
              >
                Username:qwe
              </text>
            </view>
            <text
              style="color: rgb(61, 50, 45); font-size: 18px;"
            >
              Description:Lab2 code
            </text>
          </rctsafeareaview>
        </view>
        <view>
          <rctsafeareaview
            style="background-color: rgb(243, 227, 194); flex-direction: column; margin-bottom: 5px; margin-top: 5px;"
          >
            <text
              style="color: rgb(142, 25, 18); font-size: 18px;"
            >
              Repo name:Lab3
            </text>
            <view
              focusable="true"
              style="opacity: 1;"
            >
              <text
                style="color: rgb(113, 22, 116); font-size: 18px;"
              >
                Username:haha
              </text>
            </view>
            <text
              style="color: rgb(61, 50, 45); font-size: 18px;"
            >
              Description:Fall 2018 Lab3 starter code.
            </text>
          </rctsafeareaview>
        </view>
        <view
          style="height: 0px;"
        />
      </view>
    </rctscrollview>
  </rctsafeareaview>
</DocumentFragment>
`;

exports[`Matches profile screen under error snapshot 1`] = `
<DocumentFragment>
  <rctsafeareaview
    style="flex: 1; background-color: rgb(214, 220, 231); justify-content: center; align-items: center;"
  >
    <text>
      Not Found
    </text>
  </rctsafeareaview>
</DocumentFragment>
`;

exports[`Matches repositories screen under error snapshot 1`] = `
<DocumentFragment>
  <rctsafeareaview
    style="flex: 1; background-color: rgb(214, 220, 231); justify-content: center; align-items: center;"
  >
    <text>
      Not Found
    </text>
  </rctsafeareaview>
</DocumentFragment>
`;
